probeinfo_api.yaml

```diff
@@ -604,6 +596,38 @@ components:
       default: false
       description: Set to `true` if the repo corresponds to a deprecated product or library.
 
+    LibraryName:
+      type: string
+      description: |
+        A unique identifier for this library, applicable across all variants of it.
+      example: glean-core
+
+    LibraryVariants:
+      type: array
+      description: |
+        Variants of this library.
+      items:
+        type: object
+        additionalProperties: false
+        required:
+          - dependency_name
+          - v1_name
+        properties:
+          v1_name:
+            $ref: "#/components/schemas/V1Name"
+          description:
+            $ref: "#/components/schemas/Description"
+          deprecated:
+            $ref: "#/components/schemas/DeprecatedBool"
+          branch:
+            $ref: "#/components/schemas/RepoBranch"
+          metrics_files:
+            $ref: "#/components/schemas/MetricsFiles"
+          ping_files:
+            $ref: "#/components/schemas/PingFiles"
+          dependency_name:
```

Comment:
We do not currently have any libraries that define multiple dependencies, but it's my understanding that the intention of having `library_names` be an array is that we could have a glean library that is used by multiple platforms, and so would export multiple names for multiple platforms/build systems. So this should remain plural as `dependency_names` and be an array.

Does that match your understanding, @fbertsch ?

Reply:
Hmm, I'm trying to decide. This may be a leftover from the days of when we were going to try and resolve application dependencies from the repository itself, in which case we would need to know that a single library could be referred to in different ways.

However, we've not taken that approach, and I doubt we ever will. If that's the case we get to say what the dependency names are, and they are really only used by this file. We should then be fine to unite them into one.

Reply:
So it sounds like you're advocating for not having build-system-specific names at all. In which case the applications would list dependencies by `v1_name` (in which case we should not put v1 in the name, but rather use something like `library_variant_id`). This would be somewhat difficult from a backwards-compatibility standpoint, since we'd still need to maintain build-system-specific names to inject into the v1 output.

For the scope of this PR, then, maybe it's fine to keep `dependency_name` as it is defined here.